src/protocol/clock/extentmaker.rs

```diff
@@ -0,0 +1,88 @@
+use std::num::TryFromIntError;
+use std::time::Duration;
+
+use super::extent::{Extent, ExtentIncrement, ExtentIncrementsCount, IncrementCounter};
+use super::{ClockType, StoppedClock, TimeNow, WorkingClock};
+
+pub trait ExtentFromClock<T>: Sized
```

Comment:
@da2ce7 I think it's a good name, but the first time I read `ExtendFromClock` it seemed you were going to create a child trait for another clock trait. In general, "Extent" and "Extend" are very similar for me, maybe because I'm not a native English speaker.

Reply:
Renamed to `MakeTimeExtent<Clock>`, that I hope is more intuitive.